src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -641,7 +641,7 @@ private static SuggestedActionSetPriority GetSuggestedActionSetPriority(CodeActi
 
                 if (workspace.Options.GetOption(EditorComponentOnOffOptions.CodeRefactorings) &&
                     _owner._codeRefactoringService != null &&
-                    supportsFeatureService.SupportsRefactorings(document) &&
+                    supportsFeatureService.SupportsRefactorings(_subjectBuffer) &&
```

Comment:
so... it feels like thsi could just be simple extension methods on ITExtBuffer.  i.e. ITextBuffer.SupportsRefactoring.  The extension would work by trying to find the associated workspace, then getting the service, then querying it.